src/persist/src/operators/stream.rs

```diff
@@ -876,6 +874,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
```

Comment:
@danhhz you said that you have opinions on how this should be done.

Reply:
I think it's hard (or impossible) to do it "from the outside". If we push all the `Seal` to one channel we loose the information about where they are from, if we use two channels, we lose the ordering information.

Internally, we have `SegNo`s that tell us the order in which we apply operations. We can verify that we get the expected ordering of seal calls, for the to streams, but I don't know how easy it is to get at these from the outside.

Reply:
we could do it with 3 channels and an async background task. each listen would get a channel, then the task would repeatedly read from whichever channel has something available and write it to the third channel, which would then contain exactly what the single channel here does now. it's a lot of complexity, but this test is worth it if we can't think of anything better

Reply:
is there a guarantee that the background task will read the two independent channels in the same order?

Reply:
no but that's not a requirement IMO (I'm curious if @aljoscha agrees). the important thing is that any order that we do observe upholds the invariants that are checked later in the test https://github.com/MaterializeInc/materialize/pull/9107/files#diff-484921e74ba303073fe90a8e8442998e6df16e12c8734710cd1ed3c2b4f2df3bR962-R976

Reply:
The order in which Seals from the two channels arrive would matter. Say both channels contain a `Seal(1)`, it's important to know which one was put in the channel first. And I think that information is lost.

Reply:
(Sorry if this is going very slowly, we can chat about it when we're both online at the same time)